proto/regen/ecocredit/v1alpha1/tx.proto

```diff
@@ -274,4 +280,41 @@ message MsgUpdateClassMetadata {
 }
 
 // MsgUpdateClassMetadataResponse is the MsgUpdateClassMetadata response type.
-message MsgUpdateClassMetadataResponse {}
\ No newline at end of file
+message MsgUpdateClassMetadataResponse {}
+
+message MsgSell {
+  string owner = 1;
+  repeated Order orders = 2;
+
+  message Order {
+    string batch_denom = 1;
+    string quantity = 2;
+    string ask_denom = 3;
+    string ask_price = 4;
+    bool dont_auto_retire = 5;
+  }
+}
+
+message MsgSellResponse {
+  repeated uint64 sell_order_ids = 1;
+}
+
+message MsgCancelSellOrder {
+  string owner = 1;
+  repeated uint64 sell_order_ids = 2;
+}
+
+message MsgCancelSellOrderResponse {}
+
+message MsgBuyDirect {
+  string buyer = 1;
+  repeated Order orders = 2;
+
+  message Order {
+    uint64 sell_order_id = 1;
+    string quantity = 2;
+    bool dont_auto_retire = 3;
```

Comment:
What does it mean for a buyer to provide an `auto_retire` flag? Isn't auto-retire something that would be set / determined by the seller only?

Reply:
I'm thinking we would have both default to auto-retire. Even if the seller checks `dont_auto_retire`, they'll still retire unless the buyer also checks `dont_auto_retire`. It biases the whole design towards the intended use (i.e. retirement) which I've gotten the sense is desirable